### ISSUE TITLE
Favicons are in the new cache, retrieval is to be modified.

### ISSUE THREAD

**Issue by hobgoblinsmaster** [fbarthelery/geekttrss#22]
View this [commit](https://git.tt-rss.org/fox/tt-rss.git/commit/?id=a30b9bb649d6e10a5d7c2feb73376669cf23ef68) on TTRSS sources.

The old path (`/feed-icons/<id>.ico`) is deprecated in favor of `/public.php?op=feed_icon&id=<id>&ts=<epoch>`

`ts` seems optional.

Line https://github.com/fbarthelery/geekttrss/blob/e6423dfba9f7b5b85c216256c65dfe10278a808a/app/src/main/java/com/geekorum/ttrss/network/ApiRetrofitService.kt#L119 should be something like :
```kt
it.copy(feedIconUrl = "$baseFeedsIconsUrl?op=feed_icon&${it.id}&ts=${timestamp}")
```

**Comment by fbarthelery** [comment]
Thanks for the heads up. I'll take a look once I updated my server

**Comment by fbarthelery** [comment]
Thanks for the heads up. 
I'm not sure how to properly handle with backward compatibility as this change is fairly recent.

Also, this change shouldn't have a lot of impact as we update the favicon later if we can find a better one using [favIKonSnoop](https://github.com/fbarthelery/geekttrss/tree/master/faviKonSnoop)

**Comment by hobgoblinsmaster** [comment]
Thank you to have looked a fast as you done this issue.
I'm seeing two ways to detect the new behavior. First of all the content of `$baseFeedsIconsUrl` was a path, it's now ending with `public.php`, then using the old method on a new server results in a 404 error.
The fallback with favIKonSnoop does not work very well for me (I'm seeing a very few icons).

**Comment by fbarthelery** [comment]
> First of all the content of `$baseFeedsIconsUrl` was a path, it's now ending with `public.php`, then using the old method on a new server results in a 404 error.

`$baseFeedsIconsUrl`is based on the result of the `getConfig` api call. getConfig still returns `"icons_url":"feed-icons"` and not `public.php` (at least on my server. Maybe this change in a more recent update)

>  The fallback with favIKonSnoop does not work very well for me (I'm seeing a very few icons).

I actually noticed and fixed an issue on this when I looked for the changes yesterday.

**Comment by hobgoblinsmaster** [comment]
> `$baseFeedsIconsUrl`is based on the result of the `getConfig` api call. getConfig still returns `"icons_url":"feed-icons"` and not `public.php` (at least on my server. Maybe this change in a more recent update)

Yes I'd seen this and made a big shortcut.
This is part of the [update](https://git.tt-rss.org/fox/tt-rss.git/diff/classes/rpc.php?id=a30b9bb649d6e10a5d7c2feb73376669cf23ef68), the param `icons_url` had changed :
```diff
-		$params["icons_url"] = Config::get(Config::ICONS_URL);
+		$params["icons_url"] = Config::get(Config::SELF_URL_PATH) . '/public.php';
```

**Comment by fbarthelery** [comment]
Unfortunately, this seems to only change some internal parameters and doesn't seem to be related with the `getConfig` api call. See the [API reference](https://tt-rss.org/wiki/ApiReference)

**Comment by hobgoblinsmaster** [comment]
Thanks.
Just found that explanation: https://community.tt-rss.org/t/favicons-cache-implementation-overhaul/5698. Maybe it will be corrected in the future.
Plying a bit with the api, I was not able to find a good way to discover if the old or new method is used (getVersion ?).
Waiting for a better solution from ttrss, I made a redirect in apache with success :
```apache
        RewriteEngine On
        RewriteRule ^/feed-icons/([0-9]+).ico$ /public.php?op=feed_icon&id=$1 [R]
```

**Comment by hobgoblinsmaster** [comment]
Maybe this new version of the TTRSS API may help : https://git.tt-rss.org/fox/tt-rss.git/commit/?id=a2af3a6bb4060af2903112101c9b6f4b2801b878

**Comment by fbarthelery** [comment]
It should be. I'll have a try at it on my next server update

**Comment by fbarthelery** [comment]
I just published [1.6.2](https://github.com/fbarthelery/geekttrss/releases/tag/1.6.2) which uses the new api endpoint when it's available.
Feel free to report any new issues
